service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -58,11 +62,46 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		}
 	}
 
+	var irsa bool
+	var accountID string
+	{
+		if r.provider == "aws" {
+
+			awsCluster := &v1alpha3.AWSCluster{}
+			err := r.ctrlClient.Get(ctx, types.NamespacedName{Name: cr.Name, Namespace: cr.Namespace}, awsCluster)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			if _, ok := awsCluster.Annotations[k8smetadata.AWSIRSA]; ok {
+				irsa = true
+			}
+
+			secret, err := r.k8sClient.CoreV1().Secrets(awsCluster.Spec.Provider.CredentialSecret.Namespace).Get(ctx, awsCluster.Spec.Provider.CredentialSecret.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				r.logger.Debugf(ctx, "secret '%s/%s' not found cannot set accountID", cr.Namespace, key.APISecretName(&cr))
+			} else if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			arn := string(secret.Data["aws.awsoperator.arn"])
+
+			re := regexp.MustCompile(`[-]?\d[\d,]*[\.]?[\d{2}]*`)
+			accountID = re.FindAllString(arn, 1)[0]
+
+		} else {
+			r.logger.Debugf(ctx, "not aws provider, skipping AWS IRSA check")
```

Comment:
logging this line on all providers besides `aws`, just to tell this code is not doing anything feels a bit noisy

Reply:
I removed it 👍🏻